### ISSUE TITLE
Deleting a timeout toxic can lockup proxy http requests

### ISSUE THREAD

**Issue by sunblaze** [Shopify/toxiproxy#88]
Ran into this issue in some of my tests. Here's the curl commands you can use to recreate this issue.

```
curl -i -d '{"name": "redis", "upstream": "localhost:6379", "listen": "localhost:26379"}' localhost:8474/proxies
curl -i -d '{"type":"timeout", "timeout":30000, "stream": "upstream"}' localhost:8474/proxies/redis/toxics
curl -m 1 -i 127.0.0.1:26379
curl -m 1 -i -X DELETE localhost:8474/proxies/redis/toxics/timeout
```

The final delete request seems to lock up (final command must be executed before the 30 second timeout on the toxic), and any future requests like the following no longer returns before timing out

```
curl -i localhost:8474/proxies
```

But the version request works fine still `curl -i localhost:8474/version` returns 2.0.0rc1

@Sirupsen @xthexder

**Comment by Sirupsen** [comment]
Oh, interesting. Thanks for the detailed bug report @sunblaze

**Comment by xthexder** [comment]
I took a stab at this and found out what was going on. I wrote a fix, but in the process of testing found another bug with creating a new timeout toxic. There's some strange edge case with redis closing the connection and a toxic being added at the same time.

I should have a PR ready to fix this soon.

**Comment by Sirupsen** [comment]
Fixed in 2.0
